src/test/ui/editions/edition-deny-async-fns-2015.rs

```diff
@@ -0,0 +1,42 @@
+// edition:2015
+
+#![feature(futures_api)]
+
+async fn foo() {} //~ ERROR `async fn` is not permitted in the 2015 edition
+                  //~^ ERROR async fn is unstable
+
+fn baz() { async fn foo() {} } //~ ERROR `async fn` is not permitted in the 2015 edition
+                                  //~^ ERROR async fn is unstable
+
+async fn async_baz() { //~ ERROR `async fn` is not permitted in the 2015 edition
+                       //~^ ERROR async fn is unstable
+
+    async fn bar() {} //~ ERROR `async fn` is not permitted in the 2015 edition
+                      //~^ ERROR async fn is unstable
+}
+
+struct Foo {}
+
+impl Foo {
+    async fn foo() {} //~ ERROR `async fn` is not permitted in the 2015 edition
+}
+
+trait Bar {
+    async fn foo() {} //~ ERROR `async fn` is not permitted in the 2015 edition
+                      //~^ ERROR trait fns cannot be declared `async`
+}
+
+
+
+fn main() {
+    macro_rules! accept_item { ($x:item) => {} }
+
+    accept_item! {
+        async fn foo() {}
```

Comment:
This should be an error; `async fn` shouldn't be a valid `item` at all and should be syntactically rejected on 2015.

Reply:
Okay, I think I have a fix, I'm just testing it now

Reply:
I've got the to error correctly now